src/tls/connection.c

```diff
@@ -790,10 +791,18 @@ connection_crypto_init (const char *certfile,
   if (ret != GNUTLS_E_SUCCESS)
     errx (EXIT_FAILURE, "gnutls_certificate_allocate_credentials failed: %s", gnutls_strerror (ret));
 
+  /* check if we have a separate key file */
+  keyfile = cockpit_certificate_key_path (certfile);
   if (keyfile)
     ret = gnutls_certificate_set_x509_key_file (parameters.x509_cred, certfile, keyfile, GNUTLS_X509_FMT_PEM);
-  else
-    ret = set_x509_key_from_combined_file (parameters.x509_cred, certfile);
+
+  /* fall back to combined file */
+  if (!keyfile || ret == GNUTLS_E_FILE_ERROR)
```

Comment:
This is a little bit gross...

Again: is it actually possible that `keyfile` will be NULL here?  Do we want to do something meaningful in that case, still?

Reply:
Changed so that it can't be NULL any more.